File: Assignment-1/Approach-1/MethodOne.java
import java.util.Scanner;

class MethodOne {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    double f;
    double pi = 1;

    System.out.println("Enter the number of iterations: ");
    int n = sc.nextInt();

    for(int i = n; i > 1; i--) {
      f = 2;
      for(int j = 1; j < i; j++){
         f = 2 + Math.sqrt(f);
      }
      f = Math.sqrt(f);
      pi = pi * f / 2;
    }
    pi *= Math.sqrt(2) / 2;
    pi = 2 / pi;

    System.out.println("Aproximated value of PI= " +pi);
  }
}

File: Assignment-1/Approach-2/MethodOne.py
import math

pi = 1
n = int(input("Enter the number of iterations: "))
for i in range(n , 0, -1):
	if i > 1:
		f = 2
		for j in range(1,i):
			if j < i:
				f = 2 + math.sqrt(f)

		f = math.sqrt(f)
		pi = pi * f / 2

pi *= math.sqrt(2) / 2
pi = 2 / pi

print("Approximated value of Pi= "+str(pi))
